File: Count Word/WordFrequencyCounter.java

import java.util.*;
import java.util.stream.Collectors;

public class WordFrequencyCounter {
    public static void main(String[] args) {
        String text = "Java is awesome! Java streams are powerful. Streams make Java awesome.";
        String[] words = text.replaceAll("[^a-zA-Z ]", "").split("\\s+");
        // Count the frequency of each word (case-insensitive)
        Map<String, Long> wordCounts = Arrays.stream(words)
                .collect(Collectors.groupingBy(String::toLowerCase, Collectors.counting()));

        // Display the most frequent words and their frequencies
        wordCounts.entrySet().stream()
                .sorted(Map.Entry.comparingByValue(Comparator.reverseOrder())) // Sort by frequency (desc)
                .limit(10) // Change the limit as needed
                .forEach(entry -> System.out.println(entry.getKey() + " -> " + entry.getValue()));
    }
}

File: Count Word/WordFrequencyCounter.scala
object WordFrequencyCounter extends App {
  val content = "Scala is awesome! Scala streams are powerful. Streams make Scala awesome."
  
  val wordCounts = content.toLowerCase
    .replaceAll("[^a-zA-Z\\s]", "") // Remove punctuation
    .split("\\s+") // Split by spaces
    .groupBy(identity) // Group by word
    .mapValues(_.length) // Count occurrences

  // Display the most frequent words and their frequencies
  wordCounts.toSeq
    .sortBy(-_._2) // Sort by frequency (descending order)
    .take(10) // Limit to top 10
    .foreach { case (word, count) => println(s"$word -> $count") }

}
